File: test/Data.Analyzers.Vsix.Test/CSharp/DuplicateModelAttribute.cs
﻿using DevZest.Data.Annotations;

namespace DevZest.Data.Analyzers.Vsix.Test.CSharp
{
    [CheckConstraint(nameof(CK_AlwaysTrue), "CK")]
    [CheckConstraint(nameof(CK_AlwaysTrue), "CK")]
    public class DuplicateModelAttribute : Model
    {
        [_CheckConstraint]
        private _Boolean CK_AlwaysTrue
        {
            get { return _Boolean.Const(true); }
        }
    }
}

File: test/Data.Analyzers.Vsix.Test/VisualBasic/DuplicateModelAttribute.vb
﻿Imports DevZest.Data.Annotations

<CheckConstraint(DuplicateModelAttribute._CK_AlwaysTrue, "CK")>
<CheckConstraint(DuplicateModelAttribute._CK_AlwaysTrue, "CK")>
Public Class DuplicateModelAttribute
    Inherits Model

    Friend Const _CK_AlwaysTrue = NameOf(CK_AlwaysTrue)
    <_CheckConstraint>
    Private ReadOnly Property CK_AlwaysTrue As _Boolean
        Get
            Return _Boolean.Const(True)
        End Get
    End Property
End Class
